src/main/java/seedu/address/logic/parser/SortCommandParser.java

```diff
@@ -12,20 +24,51 @@ public class SortCommandParser implements Parser<SortCommand> {
     /**
      * Parses the given {@code String} of arguments in the context of the SortCommand
      * and returns an SortCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public SortCommand parse(String args) throws ParseException {
-        Boolean isReverse = args.trim().charAt(0) == '!';
+        List<PrefixArgument> argList =
+            tokenizeToList(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+        boolean hasArguments = argList.size() != 1;
+
+        if (!hasArguments) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        List<SortArgument> sortArgs = convertPrefixArgToSortArg(argList);
+        return new SortCommand(sortArgs);
+    }
 
-        Prefix parameter = isReverse ? new Prefix(args.trim().substring(1)) : new Prefix(args.trim());
-        return new SortCommand(parameter, isReverse);
+    /**
+     * Convert a list of PrefixArgument to a list of SortArgument.
+     * First dummy PrefixArgument in the list, which represents the preamble, is excluded.
+     *
+     * @param argList List of PrefixArgument
+     * @return        List of SortArgument
+     */
+    private List<SortArgument> convertPrefixArgToSortArg(List<PrefixArgument> argList) {
+        return argList.subList(1, argList.size()).stream().map(this::convertArguments).collect(Collectors.toList());
```

Comment:
Wow. Sure you don't want to be CS2030S TA?😂

Reply:
Are you sure you want to subject our juniors to the whims of my poor time management 😆